### PR TITLE
Basic Checks for Pull Requests

### PR DESCRIPTION
**Pull request by DanielSnellings**
This pull request will add a github "workflow" (see the 'Actions' tab above). There are currently 3 tests implemented.

1. Will everything in the repo build?
2. Does everything in the repo pass tests?
3. Automated 'go fmt' on repo

I know that all of these are not true currently so they will fail for awhile, but once everything in gonomics builds and passes tests then we will quickly be able to see whether any new pull request will break code. 

Note that failing these checks will not prevent a pull request from being merged (unless we decide to make them required). It just indicates whether they pass in the 'checks' tab of a pull request.

### CODE REVIEW

**Review by RavenYL — APPROVED** [review]

**Review by cfauci — APPROVED** [review]

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          ref: ${{ github.head_ref }}
+      - name: go fmt
+        run: go fmt ./...
+      - name: Check for modified files
+        id: git-check
+        run: echo ::set-output name=modified::$(if git diff-index --quiet HEAD --; then echo "false"; else echo "true"; fi)
+      - name: Push changes
+        if: steps.git-check.outputs.modified == 'true'
+        run: |
+          git config --global user.email "AutoFmtBot@fmtyacodeffs.com"
```
Comment (craiglowe) [inline .github/workflows/go.yml:42]: I did a quick whois on this address and it didn't bring anything up.  Is it just a dummy address that you came up with?

**Review by craiglowe — COMMENTED** [review]
Wow!  So cool!!! I don't fully understand it, but I am a HUGE fan of this.  I am happy to approve once I get a walk-through of this file.  Awesome!

File: .github/workflows/go.yml (hunk 1 above)
Comment (DanielSnellings) [reply .github/workflows/go.yml:42]: Yep. It's a required field for doing an automatic push so I just made it up.

Comment (cfauci) [reply .github/workflows/go.yml:42]: this email is my favorite thing about this, as wonderful as it is

**Review by rimangan — APPROVED** [review]

**Review by cshoben — APPROVED** [review]

**Pull request merged**